server/src/comment.ts

```diff
@@ -44,18 +44,21 @@ type UidToSocialInfo = {
   [key: string]: any;
 };
 
-const useTranslateApi = isTrue(process.env.SHOULD_USE_TRANSLATION_API);
+let POLIS_ROOT = process.env.POLIS_ROOT
+var config = require(POLIS_ROOT + 'config/config.js');
+
+const useTranslateApi = isTrue(config.get('should_use_translation_api'));
 let translateClient: any = null;
 if (useTranslateApi) {
   // Tell translation library where to find credentials, and write them to disk.
   process.env.GOOGLE_APPLICATION_CREDENTIALS = ".google_creds_temp";
   // TODO: Consider deprecating GOOGLE_CREDS_STRINGIFIED in future.
-  const creds_string = process.env.GOOGLE_CREDENTIALS_BASE64
-    ? new Buffer(process.env.GOOGLE_CREDENTIALS_BASE64, "base64").toString(
+  const creds_string = config.get('google_credentials_base64')
+    ? new Buffer(config.get('google_credentials_base64'), "base64").toString(
         "ascii"
       )
-    : (process.env.GOOGLE_CREDS_STRINGIFIED as string | NodeJS.ArrayBufferView);
-  fs.writeFileSync(process.env.GOOGLE_APPLICATION_CREDENTIALS, creds_string);
+    : (config.get('google_creds_stringified') as string | NodeJS.ArrayBufferView);
+    fs.writeFileSync(process.env.GOOGLE_APPLICATION_CREDENTIALS, creds_string);
```

Comment:
Did this one get missed, or does this have to be an env variables for some reason?

Reply:
It wasn't missed but if I changed it to use convict then (I think the E2E) tests failed. I haven't dug deeply in this issue but the difference I saw was that the `GOOGLE_APPLICATION_CREDENTIALS` is set a few lines prior to this one. I tried using the `set()` function of convict but for some reason it didn't work. Should we investigate this issue?

Reply:
What's going on here is that the JS Google Translation library expects this environment variable to point to the file that credentials get read from. This seemed like the most standard way to do things when this was implemented, but it does seem like it's possible to explicitly pass these parameters through when the API hanlde is instantiated, so I think we should go that route so that we don't have any lingering "exceptions" in the configuration process.

See: https://cloud.google.com/docs/authentication/production#passing_code

In fact, as long as there's a reasonable default here, no one should ever have to set this option, since the code is using `google_creds_stringified` to create the file.

However, ideally it would be possible to manually specify an existing `google_application_credentials` file (rather than one written based on `google_creds_stringified`) to better support running the entire system on Google Cloud infrastructure.